prusti-interface/src/environment/mod.rs

```diff
@@ -534,4 +540,45 @@ impl<'tcx> Environment<'tcx> {
     pub fn identity_substs(&self, def_id: ProcedureDefId) -> SubstsRef<'tcx> {
         ty::List::identity_for_item(self.tcx, def_id)
     }
+
+    /// Evaluates the provided [ty::Predicate].
+    /// Returns true if the predicate is fulfilled.
+    /// Returns false if the predicate is not fulfilled or it could not be evaluated.
+    pub fn evaluate_predicate(&self, predicate: ty::Predicate<'tcx>, param_env: ty::ParamEnv<'tcx>) -> bool{
+        debug!("Evaluating predicate {:?}", predicate);
+        use rustc_trait_selection::traits;
+        use crate::rustc_trait_selection::traits::query::evaluate_obligation::InferCtxtExt;
+
+        let obligation = traits::Obligation {
+            cause: traits::ObligationCause::dummy(),
+            param_env,
+            recursion_depth: 0, // Arbitrarily chosen magic number
+            predicate,
+        };
```

Comment:
```suggestion
        let obligation = traits::Obligation::new(
            traits::ObligationCause::dummy(),
            param_env,
            predicate,
        );
```

(also gets rid of the not-quite-magic magic)